### ISSUE TITLE
Tweet and YouTube embeds don't load

### ISSUE THREAD

**Issue by worldofgeese** [flathub/com.logseq.Logseq#69]
I'm running the latest 0.9.9 Flatpak on Guix System.

Tweet embeds are stuck loading
![image](https://github.com/flathub/com.logseq.Logseq/assets/59834693/c6bc7570-62e5-403b-9d5c-4672a9dfacf3)

YouTube embeds
![image](https://github.com/flathub/com.logseq.Logseq/assets/59834693/81c8f574-03c8-4bc4-8c6f-7e7d824c34c7)

Expected is
![photo_2023-06-22_09-41-48](https://github.com/flathub/com.logseq.Logseq/assets/59834693/0e8c214a-c22d-4de8-855b-79347230f4f7)

![photo_2023-06-22_09-41-54](https://github.com/flathub/com.logseq.Logseq/assets/59834693/bd2964bf-d267-4ed2-9075-ac6143bfebbb)

![photo_2023-06-22_09-41-41](https://github.com/flathub/com.logseq.Logseq/assets/59834693/283f1aee-2cd8-4a0b-b6fe-8831809c1585)

Loading rich media is supported in other Flatpaks I use like Firefox and Telegram Desktop.

**Comment by kanru** [comment]
Works fine for me on Fedora. Can you attach the logs from

```sh
flatpak run com.logseq.Logseq
```

**Comment by worldofgeese** [comment]
@kanru 

```console
[3 zypak-helper] Error retrieving version property: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.portal.Flatpak was not provided by any .service files
[3 zypak-helper] Error retrieving supports property: org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.portal.Flatpak was not provided by any .service files
[3 zypak-helper] WARNING: Unknown portal version
(rsapi) init loggers
LaunchProcess: failed to execvp:
xdg-settings
[3:0625/091354.552275:ERROR:bus.cc(398)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[3:0625/091354.552332:ERROR:bus.cc(398)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
LaunchProcess: failed to execvp:
xdg-settings
09:13:54.749 › Logseq App(0.9.9) Starting... 
09:13:54.754 › restore proxy settings {:type "system"}
09:13:54.762 › set proxy to {:type "system"}
[40:0625/091357.188679:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
09:13:57.833 › :electron.handler/watch-dir {:path "/home/worldofgeese/.logseq/config"}
09:13:57.852 › :electron.handler/unwatch-dir {:path "/home/worldofgeese/Documents/bathtub"}
09:13:57.853 › :electron.handler/watch-dir {:path "/home/worldofgeese/Documents/bathtub"}
09:14:02.715 › set sync env "prod" for f27ed7e2-0ed6-4d92-9e90-6169fee69da4
09:14:02.756 › get file meta of "/home/worldofgeese/Documents/bathtub": 15 files in 6ms
09:14:03.100 › get file meta of "/home/worldofgeese/Documents/bathtub": 15 files in 3ms
[40:0625/091403.275672:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[40:0625/091413.084325:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 3 times!
Portal call failed: The name org.freedesktop.portal.Flatpak was not provided by any .service files
Portal call failed: The name org.freedesktop.portal.Flatpak was not provided by any .service files
Portal call failed: The name org.freedesktop.portal.Flatpak was not provided by any .service files
```

**Comment by kanru** [comment]
hmm, nothing jumps out. 

Can you run

```sh
flatpak info --show-permissions com.logseq.Logseq
```

It shows

```
[Context]
shared=network;ipc;
sockets=x11;wayland;pulseaudio;
devices=dri;
filesystems=home;
```

on my system.

**Comment by worldofgeese** [comment]
@kanru 

```console
[Context]
shared=network;ipc;
sockets=x11;wayland;pulseaudio;
devices=dri;
filesystems=home;
```

**Comment by kanru** [comment]
Thanks. I guess there's some issues in your system.

I installed the Guix 1.4.0 qemu image and installed flatpak in the vm. The flatpak run without problems in the vm.

**Comment by worldofgeese** [comment]
> Thanks. I guess there's some issues in your system.
> 
> I installed the Guix 1.4.0 qemu image and installed flatpak in the vm. The flatpak run without problems in the vm.

Wow you really went to extraordinary lengths to test! Thanks. Just to confirm, tweet and YouTube embeds did work in your case? These are Logseq Sync items but even if I deleted them and re-added they showed up in this perpetual state of loading 🤔.

**Comment by kanru** [comment]
Haha, thanks to Guix developers for making the qemu image readily available. I've also been interested in Guix for some time.

Yes, those embeds worked for me. I happened to test with the user flag so we can rule it out.

**Comment by worldofgeese** [comment]
I've sorted this out: the Flatpak required appending `dbus-launch` to get embed *and* plugins working. I changed the `.desktop` file to `Exec=flatpak run --user --branch=stable --arch=x86_64 --command=run.sh --file-forwarding com.logseq.Logseq @@u %U @@ --enable-features=WaylandWindowDecorations --ozone-platform-hint=auto` and that fixed it.

I believe you didn't encounter this because it seems to be a Guix bug that only affects GNOME desktop users.
